components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+                                                formData.$currency = componentSupport.getUserCurrency().code;
+                                            }
+                                        }
+                                    }
+
+                                }
+                            }
+                        }
+
+                        // remove empty strings
+                        if (!schemaItem.required && formData[key] === '') {
+                            delete formData[key];
+                        }
+                    } else {
+
+                        // Required, read-only value will keep its existing value since the form will not allow entry.
+                        formData[key] = (this.data && this.data[key]) || (schemaItem.required && schemaItem.default) || undefined;
+                    }
+                }
+
+            }
+
+            return formData;
+        }
+
+        /**
+         * @description Returns a JSON object containing form data
+         * @returns {Object} containing the raw schema data
+        */
+        getRawData() {
+
+            const schema = this.selectedSchema.properties;
+            const formData = {};
+
+            // go through the schema and get the form values
+            // eslint-disable-next-line no-restricted-syntax
+            for (const key in schema) {
+
+                if (key !== 'links' && key.indexOf('$') === -1 && !schema[key].readonly && Object.prototype.hasOwnProperty.call(schema, key)) {
+
+                    const item = schema[key];
+                    const element = this.querySelector(`[name="${key}"]`);
+
+                    if (element) {
+
+                        switch (item.type) {
+
+                            case 'array': {
+
+                                if (item.items.format === 'uri' || item.items.type === 'object') {
+                                    formData[key] = element.data;
+                                } else {
+                                    formData[key] = element.value;
+                                }
+
+                            } break;
+
+                            case 'boolean': {
+                                formData[key] = (element.checked);
+                            } break;
+
+                            default: {
+
+                                formData[key] = (element.value || '').trim();
+
+                                if (item.format && item.format === 'currency' && !formData.$currency) {
+                                    formData.$currency = componentSupport.getUserCurrency().code;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            return formData;
+        }
+
+        /**
+         * @description clears validation errors
+         * @returns {void} nothing
```

Comment:
this is my preferred way of indicating no return, just needs to be consistent across all non-returning methods